tests/main/interfaces-password-manager-service/task.yaml

```diff
@@ -0,0 +1,56 @@
+summary: Ensure that the password-manager-service interface works
+
+systems:
+    - -ubuntu-core-16-*
+
+prepare: |
+    . $TESTSLIB/pkgdb.sh
+    echo "Ensure we have a working gnome-keyring"
+    distro_install_package gnome-keyring dbus-x11
+    snap install --edge test-snapd-password-manager-consumer
+
+restore: |
+    . $TESTSLIB/pkgdb.sh
+    distro_auto_remove_packages gnome-keyring dbus-x11
+    kill $(cat dbus-launch.pid)
+    rm -f  dbus-launch.pid
+
+execute: |
+    CONNECTED_PATTERN=":password-manager-service +test-snapd-password-manager-consumer"
+    DISCONNECTED_PATTERN="(?s).*?\n- +test-snapd-password-manager-consumer:password-manager-service"
+
+    echo "Ensure things run"
+    eval $(dbus-launch --sh-syntax)
+    eval $(printf password|gnome-keyring-daemon --login)
+    eval $(gnome-keyring-daemon --start)
+    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+
+    echo "Then it is not shown as connected"
+    snap interfaces | grep -Pzq "$DISCONNECTED_PATTERN"
+
+    echo "===================================="
+
+    echo "When the plug is connected"
+    snap connect test-snapd-password-manager-consumer:password-manager-service
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
+
+    echo "Then the snap command is able use the libsecret service"
+    test-snapd-password-manager-consumer.secret-tool clear foo bar
+
+    if [ "$(snap debug confinement)" = none ] ; then
```

Comment:
@zyga - I think I did the code layout changes, but I don't know what you are referring to wrt "partial".